src/main/java/edu/kit/ipd/crowdcontrol/objectservice/payment/PaymentDispatcher.java

```diff
@@ -64,9 +61,9 @@ private void dispatchPayment(Experiment exp) {
                 .forEach((platform, paymentJobs) -> {
                     try {
                         platformManager.payExperiment(platform, exp, paymentJobs);
-                    } catch (TaskOperationException | IllegalWorkerSetException e) {
-                        //TODO maybe notify researcher?
-                        log.fatal(String.format("Unable to pay workers for platform %s", platform), e);
+                        //TODO props to the guy which missed the .join          ^^^^^^^^
+                    } catch (PreActionException e) {
+                        log.fatal("Payment of experiment "+exp+" on platform "+platform+" could not take place!", e.getCause());
```

Comment:
this came to my mind the moment after I wrote the comment, but I don't understand the difference to [this occurence](https://github.com/coolcrowd/object-service/pull/167/files#diff-2425f803cafeeac399221651ee67d14bR86)